### ISSUE TITLE
Feeding training data in batches

### ISSUE THREAD

**Issue by yankov** [fchollet/keras#146]
Is there a way to feed training data in batches? X_train can take up a lot of memory. Is it supposed to work if I split the data manually and then call model.fit on every batch?

**Comment by trungnt13** [comment]
**Is there a way to feed training data in batches?** Yes, you can
* The method _Sequential:train(X,y,accuracy)_ is used for training just 1 epoch on your own data. You can split data into small batches, then feed each batch to this method.

p/s: I don't think using model.fit in your case is good idea, since this method will train each of your batches for n_epoch and treat your single batch as entire dataset. However, I don't see early stop technique implemented in model.fit

**Comment by fchollet** [comment]
If it's a memory issue, you can just feed macrobatches (that fit in RAM, in particular in GPU memory if you're using the GPU) to `.fit()`, with `nb_epoch = 1`.

If you have your own minibatches, you can just feed them one by one to `.train()`. `train` will perform a single update of the model based on the input.

**Comment by paipai880429** [comment]
@fchollet You just mentioned that passing ```nb_epoch=1``` to the function ```model.fit(X, y)``` will resolve the memory issue, which makes me confused. Why? The training data (X, y) still needs to be loaded into memory previously at once, right ?
